src/test/scala/advancepassengerinfo/importer/processor/DqFileProcessorImplTest.scala

```diff
@@ -216,46 +218,37 @@ class DqFileProcessorTest extends TestKit(ActorSystem("MySpec"))
 
     }
 
-    "handle exceptions while getting nextFiles and start from fallbackFile" in {
-      val fallbackFileName = "drt_dq_" + yyyyMMdd(SDate.now().addDays(-1)) + "_000000_0000.zip"
-      val mockFileNamesProvider = new FileNames {
-        val fallBackListFiles: String => Future[List[String]] = fallbackFileName => Future.sequence(List(
-          Future.successful(List(fallbackFileName, "3.zip")),
-        )).map(_.flatten)
+    "handle exceptions while getting nextFiles and start again" in {
+      def createManifests(manifest: VoyageManifest): Seq[(String, Try[VoyageManifest])] = Seq(
+        ("manifest1.json", Success(manifest)),
+        ("manifest2.json", Success(manifest)),
+      )
 
-        val s3Files: String => Future[List[String]] = previous => Future.sequence(List(
+      def createMockFileNamesProvider(count: AtomicInteger): FileNames = new FileNames {
```

Comment:
Probably no value in passing in this counter. The logic is owned by the mock. You could just use a `private val isFirstCall: Boolean = true` to keep it simple and explicit. Then
```if (isFirstCall) {
  isFirstCall = false
  fail...
} else succeed...
```